### PR TITLE
chore: Release main

### PR DESCRIPTION
**Pull request by DeployDuck**
:robot: Automated Release PR

This PR was created by `release-please` to prepare the next release. Once merged:

1. A new version tag will be created
2. A GitHub release will be published
3. The changelog will be updated

Changes to be included in the next release:
---

<details><summary>training: 0.15.0</summary>

### ⚠ BREAKING CHANGES

* **training:** speedup multiscale loss ([#1212](https://github.com/ecmwf/anemoi-core/issues/1212))
* **training:** make user-specified search path take precedence and simplify Hydra search paths hierarchy ([#1181](https://github.com/ecmwf/anemoi-core/issues/1181))
* Forecast dataset dataloader ([#1133](https://github.com/ecmwf/anemoi-core/issues/1133))
* **training:** parameter freezing ([#1159](https://github.com/ecmwf/anemoi-core/issues/1159))

### Features

* Add unit check for target variables ([#1155](https://github.com/ecmwf/anemoi-core/issues/1155)) ([8f98738](https://github.com/ecmwf/anemoi-core/commit/8f98738646cea115181fb2fc2be9126bd0d99e94))
* Callback evaluate cli ([#1153](https://github.com/ecmwf/anemoi-core/issues/1153)) ([a6ed3f0](https://github.com/ecmwf/anemoi-core/commit/a6ed3f06cda5beb32a1313bf1a04b14d462008a0))
* Export JSON schema ([#1112](https://github.com/ecmwf/anemoi-core/issues/1112)) ([082f443](https://github.com/ecmwf/anemoi-core/commit/082f44343ab68995295f8d464c92bd8d12b3442d))
* Forecast dataset dataloader ([#1133](https://github.com/ecmwf/anemoi-core/issues/1133)) ([7d4b5c8](https://github.com/ecmwf/anemoi-core/commit/7d4b5c82f17ecd9ec1f49d7c54fff5bc8217d4d5))
* Make cartopy projections flexible ([#1200](https://github.com/ecmwf/anemoi-core/issues/1200)) ([979c6d5](https://github.com/ecmwf/anemoi-core/commit/979c6d57491c0e301dde1e953f65166e36e6219d))
* Model Transformation Layer — Native PipelineStage modifiers ([#410](https://github.com/ecmwf/anemoi-core/issues/410)) ([#442](https://github.com/ecmwf/anemoi-core/issues/442)) ([6dcc870](https://github.com/ecmwf/anemoi-core/commit/6dcc870e0f8984fc11840d970f14e6b78b926885))
* **models, training:** Introduce sharding support for spectral losses ([#1230](https://github.com/ecmwf/anemoi-core/issues/1230)) ([f1b09f1](https://github.com/ecmwf/anemoi-core/commit/f1b09f1e607e92a437f156002407be685a45529c))
* **models:** GT processor halo exchange ([#1162](https://github.com/ecmwf/anemoi-core/issues/1162)) ([0c0ab86](https://github.com/ecmwf/anemoi-core/commit/0c0ab86f4fa64ff8444ea6f97794fd12017c16ea))
* Projections for spectral losses ([#1087](https://github.com/ecmwf/anemoi-core/issues/1087)) ([a4ea5df](https://github.com/ecmwf/anemoi-core/commit/a4ea5df2f6281e9a706c70d379ca88c36fe31b80))
* Reuse processing callbacks ([#1213](https://github.com/ecmwf/anemoi-core/issues/1213)) ([2a72fd8](https://github.com/ecmwf/anemoi-core/commit/2a72fd81e9d78351100ed3aa783f427308d5d68c))
* **training:** Add --config-path to config validate and unify config composition ([#1216](https://github.com/ecmwf/anemoi-core/issues/1216)) ([7af92b6](https://github.com/ecmwf/anemoi-core/commit/7af92b61650f4bded5df565f198a628ca8cb6d6f))
* **training:** Add --sort option to config dump ([#1238](https://github.com/ecmwf/anemoi-core/issues/1238)) ([96d9bca](https://github.com/ecmwf/anemoi-core/commit/96d9bca43debfe9f5e0a70fe63d39bb75190bbb3))
* **training:** Ignore warmup iterations when calculating throughput ([#1254](https://github.com/ecmwf/anemoi-core/issues/1254)) ([cfd5ca2](https://github.com/ecmwf/anemoi-core/commit/cfd5ca2342d1e909bf5785c39d0356b22fbf9c86))
* **training:** More benchmark tests ([#1225](https://github.com/ecmwf/anemoi-core/issues/1225)) ([8a9f790](https://github.com/ecmwf/anemoi-core/commit/8a9f7901d2a46fde416c6ec91d08c9031787bdb9))
* **training:** Speedup multiscale loss ([#1212](https://github.com/ecmwf/anemoi-core/issues/1212)) ([bc89ef3](https://github.com/ecmwf/anemoi-core/commit/bc89ef3c58f2646ff3385fc6881946693e6ab732))
* **training:** Use local synchronization for communication groups ([#1251](https://github.com/ecmwf/anemoi-core/issues/1251)) ([b44b53b](https://github.com/ecmwf/anemoi-core/commit/b44b53bdfd66c1e4ef8d28c0109da01f23704bf0))

### Bug Fixes

* Checking variable argument name ([#1205](https://github.com/ecmwf/anemoi-core/issues/1205)) ([f0b946a](https://github.com/ecmwf/anemoi-core/commit/f0b946af876360cf913951905d87916f3654d1de))
* Config precedence ([#1218](https://github.com/ecmwf/anemoi-core/issues/1218)) ([f0e8206](https://github.com/ecmwf/anemoi-core/commit/f0e8206038a827a48b771bc6813e5bf1e972cc5e))
* Fix integration tests with torch &gt;= 2.11 ([#1187](https://github.com/ecmwf/anemoi-core/issues/1187)) ([f23c096](https://github.com/ecmwf/anemoi-core/commit/f23c096ec9091c0ec6f84c52404e84fbca8b99a4))
* Graph open dataset ([#1235](https://github.com/ecmwf/anemoi-core/issues/1235)) ([6278cf0](https://github.com/ecmwf/anemoi-core/commit/6278cf06c47f8d226ecad857bbc3185b0486724d))
* **graphs:** AICON hidden to hidden topology ([#1183](https://github.com/ecmwf/anemoi-core/issues/1183)) ([ea31de5](https://github.com/ecmwf/anemoi-core/commit/ea31de5c8df7ab4d8b99c22b303208046fd498a3))
* Remove version limit ([#1178](https://github.com/ecmwf/anemoi-core/issues/1178)) ([821b1f0](https://github.com/ecmwf/anemoi-core/commit/821b1f0ee73d1fb4310d3f8a679001697e2331f0))
* Replace matplotlib.get_cmap ([#1186](https://github.com/ecmwf/anemoi-core/issues/1186)) ([8b7f835](https://github.com/ecmwf/anemoi-core/commit/8b7f835c9df5a4df6e713cb3b64519bb5c81584d))
* Revert torch limit ([#1204](https://github.com/ecmwf/anemoi-core/issues/1204)) ([8308654](https://github.com/ecmwf/anemoi-core/commit/8308654424a0dca1cc28274cace3e00c21fc5088))
* Simplify per timestep metrics ([#1173](https://github.com/ecmwf/anemoi-core/issues/1173)) ([df59ece](https://github.com/ecmwf/anemoi-core/commit/df59ece54353ea52072101e7af2dbc0876079dfe))
* **training/profilers:** Ensure dir on multi GPU is created ([#1208](https://github.com/ecmwf/anemoi-core/issues/1208)) ([68de090](https://github.com/ecmwf/anemoi-core/commit/68de090adff5644a6b234919907dcc1d11e6cecd))
* **training:** Add anemoi weight averaging classes in order to work with Imputers and Scalers ([#1113](https://github.com/ecmwf/anemoi-core/issues/1113)) ([1a99ae0](https://github.com/ecmwf/anemoi-core/commit/1a99ae09313f3324608ab4918e0f88505c789e08))
* **training:** Allow mlflow filesystem tracking backend in tests ([#1180](https://github.com/ecmwf/anemoi-core/issues/1180)) ([de91091](https://github.com/ecmwf/anemoi-core/commit/de91091063c3b1ba51f78e4cc70a4c4e74133319))
* **training:** By default, explicitly run on gpu  ([#1209](https://github.com/ecmwf/anemoi-core/issues/1209)) ([5f0971a](https://github.com/ecmwf/anemoi-core/commit/5f0971a2422985b787d239843956c8ddd1c9adc9))
* **training:** Load model on CPU to avoid GPU OOMs ([#1145](https://github.com/ecmwf/anemoi-core/issues/1145)) ([a1036ff](https://github.com/ecmwf/anemoi-core/commit/a1036ff6e9f242394d8fc67f0be12b342144ba41))
* **training:** Make user-specified search path take precedence and simplify Hydra search paths hierarchy ([#1181](https://github.com/ecmwf/anemoi-core/issues/1181)) ([bfc10b1](https://github.com/ecmwf/anemoi-core/commit/bfc10b1677ce49309c8e6f90b5f603da5462f097))
* **training:** Parameter freezing ([#1159](https://github.com/ecmwf/anemoi-core/issues/1159)) ([4135153](https://github.com/ecmwf/anemoi-core/commit/4135153b6dea3cc8626652c6c3724cbfeff6531f))
* **training:** Persist rollout step across job resume ([#1109](https://github.com/ecmwf/anemoi-core/issues/1109)) ([46ae658](https://github.com/ecmwf/anemoi-core/commit/46ae6584d7341711c21f4179217a472d16c406d3))
* **training:** Tendency scaler timstep option fix ([#1247](https://github.com/ecmwf/anemoi-core/issues/1247)) ([70d7248](https://github.com/ecmwf/anemoi-core/commit/70d72484dfadaadf16b82f6c97cda8b567102b3e))
* Update loss schema ([#1252](https://github.com/ecmwf/anemoi-core/issues/1252)) ([d2a9639](https://github.com/ecmwf/anemoi-core/commit/d2a96397364fe445b2ca78d6cb6898b9a121aee7))
* Variable argument units change ([#1206](https://github.com/ecmwf/anemoi-core/issues/1206)) ([f2e3e82](https://github.com/ecmwf/anemoi-core/commit/f2e3e82be789bdbcd2bd49a4d6f1cc7aa71d9608))

### Documentation

* Update rollout documentation ([#1214](https://github.com/ecmwf/anemoi-core/issues/1214)) ([c3666b3](https://github.com/ecmwf/anemoi-core/commit/c3666b34133c049fb1b44aba2afafa1e3f45de2f))
</details>

<details><summary>graphs: 0.9.5</summary>

### Features

* **graphs:** Add MutualKNNEdges edge builder ([#1137](https://github.com/ecmwf/anemoi-core/issues/1137)) ([e6b8acc](https://github.com/ecmwf/anemoi-core/commit/e6b8accd56c3ce4e18439d5e0b5f81b0b54cb711))
* Projections for spectral losses ([#1087](https://github.com/ecmwf/anemoi-core/issues/1087)) ([a4ea5df](https://github.com/ecmwf/anemoi-core/commit/a4ea5df2f6281e9a706c70d379ca88c36fe31b80))

### Bug Fixes

* **graphs:** AICON hidden to hidden topology ([#1183](https://github.com/ecmwf/anemoi-core/issues/1183)) ([ea31de5](https://github.com/ecmwf/anemoi-core/commit/ea31de5c8df7ab4d8b99c22b303208046fd498a3))
* Remove version limit ([#1178](https://github.com/ecmwf/anemoi-core/issues/1178)) ([821b1f0](https://github.com/ecmwf/anemoi-core/commit/821b1f0ee73d1fb4310d3f8a679001697e2331f0))
* Revert torch limit ([#1204](https://github.com/ecmwf/anemoi-core/issues/1204)) ([8308654](https://github.com/ecmwf/anemoi-core/commit/8308654424a0dca1cc28274cace3e00c21fc5088))
* Support pyg-lib for fast graph building when using pyg &gt;= 2.8 ([#1179](https://github.com/ecmwf/anemoi-core/issues/1179)) ([c147200](https://github.com/ecmwf/anemoi-core/commit/c147200dec04cacbb413998b95ef8641f3a023c2))

### Performance Improvements

* **graphs:** Vectorize PlanarAreaWeights with shoelace area ([#1191](https://github.com/ecmwf/anemoi-core/issues/1191)) ([ab12c53](https://github.com/ecmwf/anemoi-core/commit/ab12c534f99ff05b054568b0e92b4655ccbec81a))
</details>

<details><summary>models: 0.17.0</summary>

### ⚠ BREAKING CHANGES

* **training:** speedup multiscale loss ([#1212](https://github.com/ecmwf/anemoi-core/issues/1212))

### Features

* **models:** Add wrapper class for InverseReducedSHT ([#1146](https://github.com/ecmwf/anemoi-core/issues/1146)) ([ada6641](https://github.com/ecmwf/anemoi-core/commit/ada6641b527bb33ab4755a6989f4e04536950029))
* **models:** GT processor halo exchange ([#1162](https://github.com/ecmwf/anemoi-core/issues/1162)) ([0c0ab86](https://github.com/ecmwf/anemoi-core/commit/0c0ab86f4fa64ff8444ea6f97794fd12017c16ea))
* Projections for spectral losses ([#1087](https://github.com/ecmwf/anemoi-core/issues/1087)) ([a4ea5df](https://github.com/ecmwf/anemoi-core/commit/a4ea5df2f6281e9a706c70d379ca88c36fe31b80))
* **training:** Speedup multiscale loss ([#1212](https://github.com/ecmwf/anemoi-core/issues/1212)) ([bc89ef3](https://github.com/ecmwf/anemoi-core/commit/bc89ef3c58f2646ff3385fc6881946693e6ab732))

### Bug Fixes

* Failing sht test ([#1194](https://github.com/ecmwf/anemoi-core/issues/1194)) ([133cb25](https://github.com/ecmwf/anemoi-core/commit/133cb253628c65de23a46ecfb02100b75aa8ef92))
* Missing dependency ([#1222](https://github.com/ecmwf/anemoi-core/issues/1222)) ([bb8fd10](https://github.com/ecmwf/anemoi-core/commit/bb8fd10d0c9b6135d674d9d4e411a20e4db53109))
* **models:** Fix test file discovery and add missing test coverage ([#1185](https://github.com/ecmwf/anemoi-core/issues/1185)) ([7b3b3b1](https://github.com/ecmwf/anemoi-core/commit/7b3b3b167855360faa8859c87f49d0197df1d9c0))
* **models:** Make last step euler in VectorFieldHeunSampler ([#1182](https://github.com/ecmwf/anemoi-core/issues/1182)) ([01dd751](https://github.com/ecmwf/anemoi-core/commit/01dd751e64dfd774b2a591c2fe6fd39dae06e9a7))
* **models:** Normalized relu bounding multidataset ([#1248](https://github.com/ecmwf/anemoi-core/issues/1248)) ([40bed7b](https://github.com/ecmwf/anemoi-core/commit/40bed7b407ed630ce042dde7666074c1c866469d))
* **models:** Triton GT stability ([#1172](https://github.com/ecmwf/anemoi-core/issues/1172)) ([bbca174](https://github.com/ecmwf/anemoi-core/commit/bbca17482d6176f53c16adb1d268ff1874840397))
* Remove version limit ([#1178](https://github.com/ecmwf/anemoi-core/issues/1178)) ([821b1f0](https://github.com/ecmwf/anemoi-core/commit/821b1f0ee73d1fb4310d3f8a679001697e2331f0))
* Revert torch limit ([#1204](https://github.com/ecmwf/anemoi-core/issues/1204)) ([8308654](https://github.com/ecmwf/anemoi-core/commit/8308654424a0dca1cc28274cace3e00c21fc5088))
</details>

---
> [!IMPORTANT]
> Please do not change the PR title, manifest file, or any other automatically generated content in this PR unless you understand the implications. Changes here can break the release process.
> 
> :warning: Merging this PR will:
> - Create a new release
> - Trigger deployment pipelines
> - Update package versions

 **Before merging:**
 - Ensure all tests pass
 - Review the changelog carefully
 - Get required approvals

 [Release-please documentation](https://github.com/googleapis/release-please)

### CODE REVIEW

**Review by anaprietonem — APPROVED** [review]

**Comment by DeployDuck** [comment]
🤖 Created releases:

- [training-0.15.0](https://github.com/ecmwf/anemoi-core/releases/tag/training-0.15.0)
- [graphs-0.9.5](https://github.com/ecmwf/anemoi-core/releases/tag/graphs-0.9.5)
- [models-0.17.0](https://github.com/ecmwf/anemoi-core/releases/tag/models-0.17.0)

:sunflower:
